plink_clump_hail.py

```diff
@@ -0,0 +1,366 @@
+                                                   .map(lambda _: hl.null(ht.row_field_name_outer[i].row_field_name.dtype.element_type)),
+                                                   ht.row_field_name_outer[i].row_field_name),
+                                           hl.range(hl.len(ht.global_field_name_outer))))
+    ht = ht.transmute_globals(inner_global=hl.flatmap(lambda x: x.global_field_name, ht.global_field_name_outer))
+    mt = ht._unlocalize_entries('inner_row', 'inner_global', globals_for_col_key)
+    return mt
+
+def join_clump_hts(pop, not_pop, high_quality=False, overwrite=False):
+    r'''
+    Wrapper for mwzj
+    '''
+    pop = pop.upper()
+    pheno_manifest = hl.import_table(
+    #            get_pheno_manifest_path(), 
+            'gs://ukb-diverse-pops/ld_prune/phenotype_manifest.tsv.bgz', # hardcoded path to avoid having to change user-pays
+            impute=True, 
+            key=ukb_common.PHENO_KEY_FIELDS
+            )
+    pheno_manifest = pheno_manifest.annotate(pheno_id = pheno_manifest.filename.replace('.tsv.bgz',''))
+    
+    clump_results_dir = f'{ldprune_dir}/results{"_high_quality" if high_quality else ""}/{"not_" if not_pop else ""}{pop}'
+    ls = hl.hadoop_ls(f'{clump_results_dir}/*')
+    all_hts = [x['path'] for x in ls if 'clump_results.ht' in x['path']]
+    
+    temp_dir = f'gs://ukbb-diverse-temp-30day/nb-temp/{"not_" if not_pop else ""}{pop}{"-hq" if high_quality else ""}'
+    globals_for_col_key = ukb_common.PHENO_KEY_FIELDS
+    mt = mwzj_hts_by_tree(all_hts=all_hts,
+                         temp_dir=temp_dir,
+                         globals_for_col_key=globals_for_col_key)
+#    mt = resume_mwzj(temp_dir=temp_dir, # NOTE: only use if all the temp hts have been created
+#                     globals_for_col_key=globals_for_col_key)
+
+    mt.write(get_clumping_results_path(pop=pop,
+                                       not_pop=not_pop,
+                                       high_quality=high_quality), 
+             overwrite=overwrite)
+
+def munge_mt(pop, not_pop, high_quality, parts=None):
+    r'''
+    For processing MTs before joining into the full clump mt
+    '''
+    pop_array = [p for p in all_pops if p!=pop] if not_pop else [pop]
+    mt = hl.read_matrix_table(path=get_clumping_results_path(pop=pop,not_pop=not_pop,high_quality=high_quality),
+                              _intervals=parts) # alternative path for results that are filtered to high quality variants before clumping
+    print(f'{"not_" if not_pop else ""}{pop}: {mt.count_cols()}')
+    mt = mt.annotate_cols(clump_pops = hl.literal(pop_array))
+    if 'idx' in mt.entry.keys():
+        mt = mt.select_entries(plink_clump=mt.entry.drop('idx')) # needed to clean up not_EAS and not_MID MTs (8/4/20)
+    else:
+        mt = mt.select_entries(plink_clump=mt.entry)
+    mt.describe()
+    return mt
+
+def make_single_pop_clump_mt(high_quality=False, overwrite=False):
```

Comment:
Is this the same as the function below but without the loop over not_pop? If so, you may want to consolidate